wgengine/magicsock/magicsock_linux.go

```diff
@@ -213,6 +213,19 @@ func (c *Conn) receiveDisco(pc net.PacketConn) {
 			// Too small to be a valid UDP datagram, drop.
 			continue
 		}
+
+		dstPort := binary.BigEndian.Uint16(buf[2:4])
+
+		// TODO(raggi): unsure if these are memoized by pcon, if not we should do so.
+		v4Port := c.pconn4.LocalAddr().Port
+		v6Port := c.pconn6.LocalAddr().Port
+
+		if dstPort != uint16(v4Port) && dstPort != uint16(v6Port) {
```

Comment:
I assume this is temporary and you'll actually only check the v4port for IPv4 srcs and v6Port for IPv6 srcs?

Reply:
(that is, this will need to move after the srcIP initialization below)

Reply:
Yes, this is addressed. I addressed it slightly differently, by passing in the family instead.